app/controllers/features_controller.rb

```diff
@@ -12,23 +12,23 @@ def embed
   end
 
   def new
-    if current_user.role != 'admin'
+    unless current_user.admin?
       flash[:warning] = 'Only admins may edit features.'
-      redirect_to '/features'
+      redirect_to features_url
```

Comment:
Is this auto-generated from the routes, then? Thanks!

Reply:
Yes, `features_url` and `features_path`. The first is recommended for redirection as in contains the full URL.

Reply:
I want to add that I rarely come across named routes in the codebase. And I don't see Rails' `link_to` method. Is there a specific reason for this?

Reply:
In general, I've felt that it is more readable for people who just do HTML (which is a lot of people) to just use `<a>` tags, and the `_url` Rails helpers are a little mysterious to people. I can see the advantage to having managed routes, but I think there's something good about having a literal link where you can see the URL, in a standard HTML format. 

But I'm open to discussion on it! What do you think?

Reply:
I see your point, esp. regarding templates. However, controllers are just Ruby so named routes would probably suit them better. The idea is to enable any future change of URLs without breaking things, like what would happen in #2484 (/blog -> /wiki/stories). But I don't want to be disruptive here. Maybe you think it's better to keep things as they are (for now at least). Or maybe we should get input from other contributors too. Let me know your thoughts and I'll be glad to revert relevant changes back if needed.

Reply:
How about in order to move forward, we stick with static string routes for now, and then as you said 

> get input from other contributors too

That sounds great. I do think we have trouble with route maintenance -- #1841 was trying to think through that one -- and very occasionally routes break since they are not thoroughly tested :-(

Reply:
Thanks for thinking this through! I agree re: controllers being in ruby, but the named routes are just so mysterious to newcomers! If we could smooth that learning curve a bit I'd be more comfortable using them.

Reply:
Yes, the routes file needs a bit of tidying up.
OK I'll use static strings for now. I agree that named routes can be a mystery for newcomers. Very relevant to our newcomer-friendly policy!
Last, this is not a promise, but I'll try to look into the routes issue and see if I can help. Anything you think you should mention?

Reply:
Could you tell me why the current time is (sometimes) appended to the redirect URL?